src/routes/NewBoard/NewBoard.tsx

```diff
@@ -1,26 +1,118 @@
+import {TextInputLabel} from "components/TextInputLabel";
+import {TextInput} from "components/TextInput";
+import {Button} from "components/Button";
+import {ScrumlrLogo} from "components/ScrumlrLogo";
+
+import {PassphraseModal} from "components/PassphraseDialog/PassphraseModal/PassphraseModal";
+import {Column, importBoard, Note, Participant, Voting} from "store/features";
+import {useAppDispatch} from "store";
 import {columnTemplates} from "./columnTemplates";
-import {TextInputLabel} from "../../components/TextInputLabel";
-import {TextInput} from "../../components/TextInput";
-import {Button} from "../../components/Button";
-import {ScrumlrLogo} from "../../components/ScrumlrLogo";
+import {Toast} from "../../utils/Toast";
 
 export const NewBoard = () => {
   const {t} = useTranslation();
+  const dispatch = useAppDispatch();
   const navigate = useNavigate();
   const [boardName, setBoardName] = useState<string | undefined>();
   const [columnTemplate, setColumnTemplate] = useState<string | undefined>(undefined);
   const [accessPolicy, setAccessPolicy] = useState(0);
   const [passphrase, setPassphrase] = useState("");
   const [extendedConfiguration, setExtendedConfiguration] = useState(false);
+  const [importFile, setImportFileConfiguration] = useState(false);
+  const [showPasswordModal, setShowPasswordModal] = useState(false);
+  const [loadedFile, setFile] = useState<File>();
+  const [completeBoard, setImportBoard] = useState<BoardContent>();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  type BoardContent = {
+    board: {
+      name: string;
+      description?: string;
+      accessPolicy: string;
+      passphrase?: string;
+    };
+    columns: Column[];
+    notes: Note[];
+    participants: Participant;
+    voting: Voting;
+  };
+
+  const handleDragOver = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+  };
+
+  const handleDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+  };
+
+  const closeModal = () => {
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+    setFile(new File([], ""));
+    setAccessPolicy(0);
+    setPassphrase("");
+    setShowPasswordModal(false);
+  };
 
-  async function onCreateBoard() {
+  const handleFileEvent = (event: React.ChangeEvent<HTMLInputElement> | React.DragEvent<HTMLDivElement>) => {
+    let file: File | null = null;
+    if ("dataTransfer" in event) {
+      event.preventDefault();
+      file = event.dataTransfer?.files?.[0] || null;
+    } else if ("target" in event) {
+      file = event.target.files?.[0] || null;
+    }
+
+    if (!file) return;
+
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      const content = e.target?.result as string;
+      try {
+        const data = JSON.parse(content) as BoardContent;
+        if (data.board.accessPolicy === "BY_PASSPHRASE") {
+          setShowPasswordModal(true);
+          setAccessPolicy(1);
```

Comment:
instead of `1`, use `AccessPolicy.BY_PASSPHRASE`

Reply:
What happens if the value for accessPolicy is `AccessPolicy.BY_INVITE`? can you set the access policy directly to the value of `data.board.accessPolicy`?

Reply:
If accessPolicy has the value `AccessPolicy.BY_INVITE` it will be imported as a board with this accessPolicy. The reason I am checking if the policy is set to `PASSPHRASE` is to open the PasswordModal